messenger.go

```diff
@@ -916,6 +935,111 @@ func (p *postProcessor) matchMessage(message *protocol.Message, chats []*Chat) (
 	}
 }
 
+type EnsDetails struct {
+	Name      string `json:"name"`
+	PublicKey string `json:"publicKey"`
+}
+
+type EnsResponse struct {
+	Name            string           `json:"name"`
+	Verified        bool             `json:"verified"`
+	Error           error            `json:"error"`
+	PublicKey       *ecdsa.PublicKey `json:"-"`
+	PublicKeyString string           `json:"publicKey"`
+}
+
+func (m *Messenger) verifyEnsName(ensInfo EnsDetails, ethclient *ethclient.Client, output chan EnsResponse) {
+	publicKeyStr := ensInfo.PublicKey
+	ensName := ensInfo.Name
+	m.logger.Info("Resolving ENS name", zap.String("name", ensName), zap.String("publicKey", publicKeyStr))
+	response := EnsResponse{
+		Name:            ensName,
+		PublicKeyString: publicKeyStr,
+	}
+
+	expectedPubKeyBytes, err := hex.DecodeString(publicKeyStr)
+	if err != nil {
+		response.Error = err
+		output <- response
+		return
+	}
+
+	publicKey, err := crypto.UnmarshalPubkey(expectedPubKeyBytes)
+	if err != nil {
+		response.Error = err
+		output <- response
+		return
+	}
+
+	// Resolve ensName
+	resolver, err := ens.NewResolver(ethclient, ensName)
+	if err != nil {
+		m.logger.Error("error while creating ENS name resolver", zap.String("ensName", ensName), zap.Error(err))
+		response.Error = err
+		output <- response
+		return
+	}
+	x, y, err := resolver.PubKey()
+	if err != nil {
+		m.logger.Error("error while resolving public key from ENS name", zap.String("ensName", ensName), zap.Error(err))
+		response.Error = err
+		output <- response
+		return
+	}
+
+	// Assemble the bytes returned for the pubkey
+	pubKeyBytes := elliptic.Marshal(crypto.S256(), new(big.Int).SetBytes(x[:]), new(big.Int).SetBytes(y[:]))
+
+	response.PublicKey = publicKey
+	response.Verified = bytes.Equal(pubKeyBytes, expectedPubKeyBytes)
+	output <- response
+}
+
+// VerifyENSName verifies that a registered ENS name matches the expected public key
+func (m *Messenger) VerifyENSNames(ensDetails []EnsDetails, rpcEndpoint, contractAddress string) (map[string]EnsResponse, error) {
```

Comment:
```suggestion
func (m *Messenger) VerifyENSNames(rpcEndpoint, contractAddress string, ensDetails []EnsDetails) (map[string]EnsResponse, error) {
    // Can address and RPC endpoint have some defaults? Does it change during runtime? 
    verifier := ENSVerifier(WithContractAddress(contractAddress), WithRPCEndpoint(rcpEndpoint))
    response, err := verifier.CheckBatch(ensDetails)
    ...
}
```

We store too much logic in `Messenger`. I recommend creating `ENSVerifier` in `ens.go` and here we can define a single `VerifyENSNames` method.